src/main/java/Sorting.java

```diff
@@ -0,0 +1,273 @@
+
+
+import java.util.*;
+
+/**
+ * Created by sam on 6/17/16.
+ */
+public class Sorting {
+
+    // implement bubble sort
+    /*
+        input heapArray of sortable items?
+         ints for now
+         while unsorted
+          loop thru entire heapArray and if an element is less than the previous element then swap them
+          if looped thru the entire heapArray and every element is >= previous, sorted
+     */
+
+    public static List<Integer> bubbleSort(List<Integer> array) throws IllegalArgumentException {
+
+        if(array == null) {
+            throw new IllegalArgumentException("invalid input array to bubble sort: input array is null");
+        }
+
+        if(array.isEmpty()) {
```

Comment:
Is this really worthy of an IllegalArgumentException? Feels like an potential optimization to just return an empty array...